pypesto/visualize/sampling.py

```diff
@@ -351,7 +351,10 @@ def get_data_to_plot(
     theta_lb = result.problem.lb
     theta_ub = result.problem.ub
 
-    param_names = result.problem.x_names
+    # get parameter names from all non-fixed parameters
+    to_exclude = result.problem.x_fixed_indices
+    all_parameters = result.problem.x_names
+    param_names = [element for i, element in enumerate(all_parameters) if i not in to_exclude]
```

Comment:
should be possible via
```python
param_names = problem.get_reduced_vector(problem.x_names)
```
, please check

Reply:
indeed, thanks!